### PR TITLE
Move remaining tests with NLL differences to revisions

### PR DESCRIPTION
**Pull request by jackh726**
Based on #97206

I've already filed issues for any important differences that I've spotted: #97252 #97253 #97256 #97267

There is a lot here, but each commit is self-contained as a separate directory. I can split into separate PRs as wanted or needed.

### CODE REVIEW

**Comment by jackh726** [comment]
@cjgillot while this is technically blocked on #97206, each commit is independent (including from that). I decided to just use that as a base, since it modifies NLL tests. It's also already approved and in the bors queue . So, I would appreciate a review whenever you have time (I wouldn't mind just being able to just have this approved for when the other lands, for example).

**Comment by jackh726** [comment]
Note to self, need to update this once #96515 lands (https://github.com/rust-lang/rust/pull/96515/files#diff-6a6b025c2f35f56e17f04a29082743bcecc56b4ec83a27cbea3b676c9369dea6R1)

File: src/test/ui/borrowck/two-phase-nonrecv-autoref.rs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// revisions: nll
+// revisions: base nll
 //[nll]compile-flags: -Z borrowck=mir
 
 //[g2p]compile-flags: -Z borrowck=mir -Z two-phase-beyond-autoref
```
Comment (cjgillot) [inline src/test/ui/borrowck/two-phase-nonrecv-autoref.rs:4]: Does this option still exist in the compiler?

**Comment by cjgillot** [comment]
@jackh726 are there cases you want to draw attention to?  I did not really anything stand out, the changes seem mostly mechanical.
r=me once https://github.com/rust-lang/rust/pull/97206 lands

File: src/test/ui/borrowck/two-phase-nonrecv-autoref.rs (hunk 1 above)
Comment (jackh726) [reply src/test/ui/borrowck/two-phase-nonrecv-autoref.rs:4]: I'm not sure - preexisting  thought and I'd rather not make this PR anything other than mechanical. (Though I should try to remember to come back to this at some point)

**Comment by jackh726** [comment]
Nope, no cases to draw attention to. I don't think there was anything really surprising in the nll diagnostics compared to base. Obviously, I already filed issues for the cases where we obviously are slightly worse. But there was nothing that is like "oh, that's not good".

**Comment by jackh726** [comment]
@bors r=cjgillot p=1

(bitrotty)

**Pull request merged**